source/ShopPanel.h

```diff
@@ -104,7 +104,7 @@ class ShopPanel : public Panel {
 	virtual bool Release(int x, int y) override;
 	virtual bool Scroll(double dx, double dy) override;
 
-	int64_t LicenseCost(const Outfit *outfit, bool onlyOwned = false) const;
+	int64_t LicenseCost(const Outfit *outfit, bool onlyOwned = false, const Ship *ship = nullptr) const;
```

Comment:
isnt it either the outfit or the ship?

Reply:
Indeed, but this is ShopPanel, which gets called from both the outfitter and the shipyard, so it has to be able to handle both.  When it's called from the outfitter, outfit is an outfit.  When it's called from the shipyard, outfit is actually a list of attributes the ship has.  Since that's not something that's going to be found in the player's list of stock outfits, we need a separate check to see if the ship is in the player's list of stock ships.

Reply:
Alternatively, you could make a `LicenseCost(const Ship *ship, bool onlyOwned = false)` method that would return 0 if the ship is in stock, and if not, it would return LicenseCost for the ship's attributes.
Maybe this would show clearer what's going on here, but your current solution isn't bad.

Reply:
You know what, I think having two calls for this makes sense.  And one's kinda just a wrapper around the other, but the calling interface is a bit cleaner this way.

We can even get rid of onlyOwned for ships, since it's only used for buy outfits from cargo/storage.